### PR TITLE
feat(lookup): lookup-on-clear attribute

### PR DESCRIPTION
**Pull request by jnorkus**
Add a callback that is fired when lookup is cleared either by button or keyboard

### CODE REVIEW

File: src/lookup/test/lookup.spec.js [hunk 1]
```diff
@@ -316,6 +316,29 @@
         });
     });
     
+    describe('Lookup on clear callback', function() {
+        beforeEach(function() {
+            $scope.onClear = jasmine.createSpy('onClear');
+            initDirective('<div dd-lookup ng-model="model" lookup-on-clear="onClear()" url="\'/api/drivers/lookup\'"></div>');
+        });
+        
+       it('is called when clear button is pressed.', function() {
```
Comment (Anjmao) [inline src/lookup/test/lookup.spec.js]: Add test to be sure that lookup-on-clear not called on initial load when model is empty
